app/javascript/src/components/editor/CodeMirror.svelte

```diff
@@ -52,7 +55,21 @@
     })
 
     $editorStates[currentId] = createEditorState($currentItem.content)
+
+    onStateUpdateEnd()
+  }
+
+  /**
+   * This is fired after updateEditorState, after all required bits are set.
+   * Part of this is wrapped in an empty setTimeout to wait for the view state to update.
+   */
+  function onStateUpdateEnd() {
     view.setState($editorStates[currentId])
+
+    setTimeout(() => {
```

Comment:
Maybe [`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame)  could work here?

Reply:
Sure! Ultimately it's the same result, but requestAnimationFrame is a bit more explicit I suppose

Reply:
I was hoping it'd remove that frame where the scrollbar shifts after changing back to the file, as shown on your gif. I guess it did not.

Reply:
Oh you know what, it did do that!

![scroll-state-2](https://github.com/Mitcheljager/workshop.codes/assets/12848235/2687f0d7-36c2-410e-92e0-7be8f876eced)